### DISCUSSION TITLE
Ways to integrate SMS or email magic link authentication

### DISCUSSION THREAD

**Discussion by alexturpin** [supabase/supabase#2519] [Feature Requests]
Hey folks!

I wonder if there are plans for more manually setup authentication flows, to integrate with SMS and email providers to use one time codes or links? As of now, I can't seem to figure out how one would leverage Supabase auth with that. Perhaps when Workflows come along?

**Comment by alexturpin** [comment]
Oh, just saw the latest blog post as well about Auth v2 for #supalaunchweek 🤞

**Reply by inian** [reply]
Stay tuned 🤐

**Reply by joewinke** [reply]
Any update here? its been 5 years :D

**Comment by alexturpin** [comment]
One way this could work is a client-side function, `authWithOTP` or something, which calls a server-side workflow that is responsible for generating and returning a one time password, as well as any logic to notify users of it. Supabase handles storing the code for users and expiring it. Client-side function can be called again with the code (or a different one) and the user is logged in if it's the right code within the expiry.

That would be flexible enough for the community to build all kinds of integrations for the auth system.
